src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy

```diff
@@ -119,8 +136,41 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
         quiet.set(false)
         pull.set(false)
         cacheFrom.empty()
+        imageIdFile.set(new File(project.buildDir, ".docker/${path}-imageId.txt"))
+
+        Boolean upToDate = false;
+        if(imageIdFile.get().asFile.exists()){
+            String imageId = imageIdFile.get().asFile.text
+            
+            println "Looking in repository for imageId: $imageId"
+            DockerClient dockerClient = getDockerClient(createDockerClientConfig(), (DockerExtension) project.extensions.getByName(DockerRemoteApiPlugin.EXTENSION_NAME))
+            List<Image> images = dockerClient.listImagesCmd().exec()
+            for(image in images) {
+                println image.toString()
+                if(image.toString().contains(imageId)){
+                    println "Found matching imageId in docker registry"
+                    upToDate = true
+                    break;
+                }
+            }
+        }
+        outputs.upToDateWhen{upToDate}
```

Comment:
As a side note: All this logic would need to be part of the Closure body. How you have it right now won't delay the execution of the code properly meaning the extension would be evaluated too early.

Reply:
You're right later Yesterday I had this in a afterEvaluate closure and it gives me access to URL and other class members